pkg/generator/appconfiguration/generator/workload/magic_env_var.go

```diff
@@ -0,0 +1,133 @@
+package workload
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	SecretEnvParser    MagicEnvParser = NewSecretEnvParser()
+	ConfigMapEnvParser                = NewConfigMapEnvParser()
+	RawEnvParser                      = NewRawEnvParser()
+)
+
+// MagicEnvVar generates a specialized EnvVar based on the key and
+// value of environment.
+func MagicEnvVar(k, v string) *corev1.EnvVar {
+	supportedParsers := []MagicEnvParser{
```

Comment:
We can initialize `supportedParsers` outside of this function since it only needs to be initialized once.